contracts/token/ERC721/IERC721.sol

```diff
@@ -42,12 +41,37 @@ interface IERC721 is IERC165 {
      * either {approve} or {setApprovalForAll}.
      */
     function transferFrom(address from, address to, uint256 tokenId) external;
+
+    /**
+     * @dev Approves the transfer of a specific NFT (`tokenId`) to an account (`to`)
+     *
+     * Requirements:
+     * - The caller must own the token or be an approved operator
+     */
     function approve(address to, uint256 tokenId) external;
+
+    /**
+     * @dev Gets the approved account for a specific NFT (`tokenId`)
+     */
     function getApproved(uint256 tokenId) external view returns (address operator);
 
+    /**
+     * @dev Enable or disable (`_approved`) an account (`operator`) to manage all of the assets of the sender (`msg.sender`).
+     *
+     * Emits an {ApprovalForAll} event.
+     */
     function setApprovalForAll(address operator, bool _approved) external;
-    function isApprovedForAll(address owner, address operator) external view returns (bool);
 
+    /**
+     * @dev Queries if an account (`operator`) is allowed to manage all of the assets of another account (`owner`)
+     */
+    function isApprovedForAll(address owner, address operator) external view returns (bool);
 
+    /**
+     * @dev Safely transfers a specific NFT (`tokenId`) from one account (`from`) to another (`to`)
```

Comment:
I'd expand on what it means to 'safely' transfer (that a receiving contract implements IERC777Recipient and returns the correct value), and why this is important.

Reply:
According to the [ERC777 specification](https://eips.ethereum.org/EIPS/eip-777) it seems that it's only usable with ERC20 tokens. Aren't you rather referring to ERC721Receiver ?

Reply:
Yes, that was a mistake on my part :)